src/bin/cargo/cli.rs

```diff
@@ -356,7 +356,16 @@ For more information, see issue #12207 <https://github.com/rust-lang/cargo/issue
                 let global_args = GlobalArgs::new(sub_args);
                 let new_args = cli(gctx).no_binary_name(true).try_get_matches_from(alias)?;
 
-                let new_cmd = new_args.subcommand_name().expect("subcommand is required");
+                let new_cmd = match new_args.subcommand_name() {
```

Comment:
nit: Could we switch to either let-else or `ok_or_else()`? That would make code a bit clearer